### PR TITLE
Make Mixxx build with Qt6

### PR DESCRIPTION
**Pull request by Holzhaus**
Makes it possible to build Mixxx with Qt6 without any compiler warnings or linker errors. Based on:
- #4620
- #4622 
- #4623

Next step is to fix the QML UI, because there are quite some incompatibilities apparently.

### CODE REVIEW

**Comment by Be-ing** [comment]
Interesting... how do you plan to remove no longer used code?

File: src/engine/enginebuffer.cpp [hunk 1]
```diff
@@ -551,7 +553,9 @@ void EngineBuffer::slotTrackLoaded(TrackPointer pTrack,
     TrackPointer pOldTrack = m_pCurrentTrack;
     m_pause.lock();
 
+#if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
```
Comment (uklotzde) [inline src/engine/enginebuffer.cpp:556]: Let's switch these code blocks, i.e. the Qt 6 variant should take precedence and be placed before the legacy workaround. Consistency, makes it easier to throw away all legacy code later.

Comment (Holzhaus) [reply src/engine/enginebuffer.cpp:556]: Done.

**Comment by Holzhaus** [comment]
> Interesting... how do you plan to remove no longer used code?

I don't have a specific plan. The immediate goal was to make Mixxx build with Qt6 because that unblocks Qt6 QML development. 

Removing everything from the old skin system at once is a huge task, and such tasks almost always lead to people giving up or being too intimidate by the size to even start working on it.

Instead, this just removes QGLWidget, the weird vsync hacks, etc. and leaves everything else intact, which keeps the PR small. Removing the legacy skin stuff from the Qt6 build can be done in smaller PRs, every at the source file level. In the worst case, we can remove it once we go Qt6-only.

**Comment by Be-ing** [comment]
I suppose procrastinating the linking problems is one way forward... I'm still puzzled how you expect to ever remove dead code without rewriting the entire library system from scratch.

**Comment by Holzhaus** [comment]
> I suppose procrastinating the linking problems is one way forward... I'm still puzzled how you expect to ever remove dead code without rewriting the entire library system from scratch.

Well, the main goal of this PR is neither removing dead code nor rewriting the library system from scratch. The goal is making the Mixxx build with Qt6. I'm not saying the library system shouldn't be refactored or rewritten, or that we shouldn't remove the dead code at some point. But there is no need to connect these barely related topics and block progress on any of then unless the others are also solved. That kind of unnecessary scope creep is exactly what we previously identified as a problem with contributing to mixxx, and what wanted to avoid in the future. It scares people away from starting to work on it, leads to huge unreviewable PRs and generally makes stuff take way longer.

Now that Mixxx builds with Qt6, we don't have to fix all these issues sequentially anymore. We can work on stuff in parallel, for example:

- @daschuer could investigate failing Qt6 tests,
- @uklotzde could start replacing the library with aoide,
- @Swiftb0y and @ronso0 could improve the QML skins and
- I could remove the old legacy skins stuff.

And even if we don't remove the legacy skin code and don't replace the library yet, we already mostly solved the issue of Qt5 becoming deprecated. So in theory we could even publish a Qt6 Mixxx release where the dead code is still build (and hopefully removed by the linker).

**Comment by Holzhaus** [comment]
OK, build succeeds but skme test fails when built with Qt6. I'll check the clazy warning, but fixing the tests is out of scope for this PR. It's a separate branch anyway.

**Comment by ywwg** [comment]
Alternatively, you could disable the broken tests with a comment // Breaks under QT6.  Unless you're worried people won't bother to fix the tests later :).  But yes this sounds like a good plan.

**Comment by Holzhaus** [comment]
> Alternatively, you could disable the broken tests with a comment // Breaks under QT6. Unless you're worried people won't bother to fix the tests later :). But yes this sounds like a good plan.

These tests fail:
```
	293 - EngineSyncTest.ZeroLatencyRateChangeQuant (Failed)
	294 - EngineSyncTest.ZeroLatencyRateDiffQuant (Failed)
	295 - EngineSyncTest.ActivatingSyncDoesNotCauseDrifting (Failed)
	302 - EngineSyncTest.SyncPhaseToPlayingNonSyncDeck (Failed)
	304 - EngineSyncTest.UserTweakPreservedInSeek (Failed)
	311 - EngineSyncTest.SeekStayInPhase (Failed)
	724 - WPushButtonTest.LongPressLatchTest (Failed)
```

The WPushButtonTest doesn't matter and can be disabled, but the other tests probably fail because of this commit: https://github.com/mixxxdj/mixxx/pull/4621/commits/18236e6a802b916041f834aa63c2862afc4ec70f

I have no idea how this complicated and weird visualplayposition/vsync code is supposed to work and how to remove it properly. 
As a hack, I just replaced it with the playposition CO. I'd appreciate if someone who is more familiar with it could have a look.

**Comment by Holzhaus** [comment]
All green. Now that the separate qt6 branch is obsolete, I'll close this and open a new PR against main.

**Pull request closed without merging**